src/main/java/backend/Logic.java

```diff
@@ -258,18 +258,20 @@ public Model getRepo(String repoId) {
         List<String> originalLabels = issue.getLabels();
 
         logger.info("Changing labels for " + issue + " on UI");
-        /* Calls models to replace the issue's labels locally since the the reference to the issue here
-           could be invalidated by changes to the models elsewhere */
-        Optional<TurboIssue> localReplaceResult =
-                models.replaceIssueLabels(issue.getRepoId(), issue.getId(), newLabels);
-        if (!localReplaceResult.isPresent()) {
+        CompletableFuture<Optional<TurboIssue>> localReplaceResult =
+                repoOpControl.replaceIssueLabelsLocally(issue, newLabels);
+        if (localReplaceResult.isCompletedExceptionally() || localReplaceResult.isCancelled()) {
             return CompletableFuture.completedFuture(false);
         }
-        refreshUI();
+        Optional<TurboIssue> locallyModifiedIssue = localReplaceResult.join();
+        if (!locallyModifiedIssue.isPresent()) {
+            return CompletableFuture.completedFuture(false);
+        }
+        localReplaceResult.thenRun(this::refreshUI);
```

Comment:
I was wondering that because right before, we `join` `localReplaceResult`, and `thenRun` on a completed future will execute the action immediately. Not sure if i'm misunderstanding.